### DISCUSSION TITLE
Can you use type III errors in Adonis2()?

### DISCUSSION THREAD

**Discussion by Rob-murphys** [vegandevs/vegan#541] [Q&A]
Firstly thanks for an obviously amazing package overall!

I am wanting to run a ANCOVA on my distance matrix to control for genome quality and am thus looking if I can use `adonis2()` for that ppurpose with the following:

`adonis2(GCF_profiles ~ metadata$Host + genome_metrics$N50, permutations = 10000, by = "terms", method = "bray")`

However I am given to understand that when conducting an ANCOVA we should be concerned with Type III errors and controlling for type I errors can give the wrong results. Normally one would do with with the `Anova()` function, like so on the `iris` dataset:
```
fit2=aov(Petal.Length~Species+Sepal.Length,df)
Anova(fit2, type="III")
```

Is there a way to change the error structure in adonis2 as Anova obviously does not accept `adonis2` objects as output.

**Comment by gavinsimpson** [comment]
Won't `by = "margin"` give you what you want?
